bin/stress-test/src/main.rs

```diff
@@ -0,0 +1,283 @@
+use rand::Rng;
+use rayon::iter::{IndexedParallelIterator, IntoParallelIterator, ParallelIterator};
+use tokio::{
+    sync::mpsc::{channel, Receiver, Sender},
+    task,
+};
+
+#[derive(Parser)]
+#[command(version)]
+pub struct Cli {
+    #[command(subcommand)]
+    pub command: Command,
+}
+
+#[derive(Subcommand)]
+pub enum Command {
+    SeedStore {
+        #[arg(short, long, value_name = "DUMP_FILE", default_value = "./miden-store.sqlite3")]
+        dump_file: PathBuf,
+
+        #[arg(short, long, value_name = "NUM_ACCOUNTS")]
+        num_accounts: usize,
+
+        #[arg(short, long, value_name = "GENESIS_FILE")]
+        genesis_file: PathBuf,
+    },
+}
+
+const BATCHES_PER_BLOCK: usize = 16;
+const TRANSACTIONS_PER_BATCH: usize = 16;
+const NOTES_PER_TRANSACTION: usize = MAX_OUTPUT_NOTES_PER_BATCH / TRANSACTIONS_PER_BATCH;
+
+/// Create and store blocks into the store. Create a given number of accounts, where each account
+/// consumes a note created from a faucet. The cli accepts the following parameters:
+/// - `dump_file`: Path to the store database file.
+/// - `num_accounts`: Number of accounts to create.
+/// - `genesis_file`: Path to the genesis file of the store.
+#[tokio::main]
+async fn main() {
+    let cli = Cli::parse();
+
+    match &cli.command {
+        Command::SeedStore { dump_file, num_accounts, genesis_file } => {
+            seed_store(dump_file, *num_accounts, genesis_file).await;
+        },
+    }
+}
+
+/// Create a new faucet account with a given anchor block.
+fn create_faucet(anchor_block: &BlockHeader) -> AccountId {
+    let coin_seed: [u64; 4] = rand::thread_rng().gen();
+    let mut rng = RpoRandomCoin::new(coin_seed.map(Felt::new));
+    let key_pair = SecretKey::with_rng(&mut rng);
+    let init_seed = [0_u8; 32];
+
+    let (new_faucet, _seed) = AccountBuilder::new(init_seed)
+        .anchor(anchor_block.try_into().unwrap())
+        .account_type(AccountType::FungibleFaucet)
+        .storage_mode(AccountStorageMode::Private)
+        .with_component(RpoFalcon512::new(key_pair.public_key()))
+        .with_component(
+            BasicFungibleFaucet::new(TokenSymbol::new("TEST").unwrap(), 2, Felt::new(100_000))
+                .unwrap(),
+        )
+        .build()
+        .unwrap();
+    new_faucet.id()
+}
+
+/// Create a new note with a single fungible asset.
+fn create_note(faucet_id: AccountId) -> Note {
+    let asset = Asset::Fungible(FungibleAsset::new(faucet_id, 10).unwrap());
+    let coin_seed: [u64; 4] = rand::thread_rng().gen();
+    let rng = RpoRandomCoin::new(coin_seed.map(Felt::new));
+    NoteBuilder::new(faucet_id, rng)
+        .add_assets(vec![asset])
+        .build(&TransactionKernel::assembler())
+        .unwrap()
+}
+
+/// Create a new account with a given public key and anchor block. Generates the seed from the given
+/// index.
+fn create_account(anchor_block: &BlockHeader, public_key: PublicKey, index: usize) -> AccountId {
```

Comment:
Instead of `usize`, I think this could be `u64` directly, unless there's something I'm missing